File: 2023 styczeń/INF.04-02-23.01-SG & INF.04-01-23.01-SD/Aplikacja Konsolowa/C++/main.cpp
#include <iostream>

using namespace std;

/*
    * klasa: Note
    * opis: klasa opisująca notatkę
    * pola: count - statyczne pole przechowujące ilośc notatek
    *       identifier - identyfikator notatki
    *       title - tytuł notatki
    *       text - treść notatki
    * autor: 00000000000
*/
class Note
{
private:
    static int count;
    int identifier;

protected:
    string title, text;

public:
    Note(string newNoteTitle, string newNoteText)
    {
        title = newNoteTitle;
        text = newNoteText;
        count++;
        identifier = count;
    }
    void displayNote()
    {
        cout << "Tytuł: " << title << endl;
        cout << "Treść notatki: " << text << endl;
    }
    void debug()
    {
        cout << "Licznik: " << count << "; Identifikator: " << identifier << "; Tytuł: ";
        cout << title << "; Treść: " << text << "; " << endl;
    }
};

int Note::count = 0;

int main()
{
    string newNoteTitle;
    string newNoteText;

    cout << "Podaj tytuł notatki: ";
    cin >> newNoteTitle;

    cout << "Podaj treść notatki: ";
    cin >> newNoteText;

    Note note1(newNoteTitle, newNoteText);

    note1.displayNote();
    note1.debug();

    cout << "\nPodaj tytuł notatki: ";
    cin >> newNoteTitle;

    cout << "Podaj treść notatki: ";
    cin >> newNoteText;

    Note note2(newNoteTitle, newNoteText);

    note2.displayNote();
    note2.debug();
}

File: 2023 styczeń/INF.04-02-23.01-SG & INF.04-01-23.01-SD/Aplikacja Konsolowa/Python/main.py
class Note:
    count = 0
    
    def __init__(self, newNoteTitle, newNoteText):
        self._title = newNoteTitle
        self._text = newNoteText
        
        Note.count += 1
        self.__identifier = Note.count
        
    def displayNote(self):
        print("Tytuł:", self._title, "\nTreść notatki:", self._text)
    
    def debug(self):
        print("Licznik:", self.count, "; Identifikator:", self.__identifier, "; Tytuł:", self._title, "; Treść:", self._text, ";")
        
newNoteTitle = input("Podaj tytuł notatki: ")
newNoteText = input("Podaj treść notatki: ")

newNote1 = Note(newNoteTitle, newNoteText)

newNote1.displayNote()
newNote1.debug()

newNoteTitle = input("Podaj tytuł notatki: ")
newNoteText = input("Podaj treść notatki: ")

newNote2 = Note(newNoteTitle, newNoteText)

newNote2.displayNote()
newNote2.debug()